get_sig_pitch.m

```diff
@@ -1,125 +1,197 @@
+%
+% The output arguments are:
+% 1.) `pitchsig`. Pitch values. Where Praat did not detect pitch, value is NaN
+% 2.) `pitch_taxis`. Vector of time points (in seconds) aligned with output arg 1
 
+if nargin < 7
+    params = struct;
+end
 if nargin < 6 || isempty(yes_verbose)
-  yes_verbose = 0;
+    yes_verbose = 0;
 end
 if nargin < 5 || isempty(yanalstep_ms)
-  yanalstep_ms = 10;
+    yanalstep_ms = 10;
 end
 if nargin < 4 || isempty(yanalframe_ms)
-  yanalframe_ms = 30;
+    yanalframe_ms = 30;
 end
 
-[rows_sig,cols_sig] = size(sig);
-yes_transpose = 0;
-if cols_sig > 1
-  if rows_sig > 1
-    error('sig cannot be a matrix');
-  else
-    yes_transpose = 1;
-  end
-end
+% these same pitch tracking settings are also stored in:
+%   free-speech\speech\get_sigproc_defaults.m 
+%   wave_viewer\wave_viewer.m\get_sigproc_params
+defaultParams.pitchlimits = [50 300];
+defaultParams.ptrack_method = 'praat';
+defaultParams.ptrack_max_candidates = 15;
+defaultParams.ptrack_pitch_very_accurate_checkbox = 'no';
+defaultParams.ptrack_silence_thresh = 0.03;
+defaultParams.ptrack_voicing_thresh = 0.45;
+defaultParams.ptrack_octave_cost = 0.01;
+defaultParams.ptrack_octave_jump_cost = 0.35;
+defaultParams.ptrack_voiced_unvoiced_cost = 0.14;
+params = set_missingFields(params, defaultParams, 0);
 
-if yes_transpose
-  sig = sig';
-end
+%%
+switch params.ptrack_method
+    case 'praat'
+        % execute Praat script using params
+        %must be in git repo folder to run praat function, so save current location and go there
+        curr_dir = pwd;
+        temp_str = which('get_pitch_tracks.praat');
+        praat_path = fileparts(temp_str);
+        cd(praat_path)
+
+        %% Praat wrapper code here
+        % write y to file (this will be deleted later, it is just written to the
+        % current directory)
+        audiowrite('temp_wav.wav',sig,fs)
+
+        if ismac
+            status = system(['"/Applications/Praat.app/Contents/MacOS/Praat" --run get_pitch_tracks.praat "' pwd '" "temp_wav" ' ...
+                num2str(params.pitchlimits(1)) ' ' num2str(params.max_candidates) ' ' params.pitch_very_accurate_checkbox ' ' ...
+                num2str(params.silence_thresh) ' ' num2str(params.voicing_thresh) ' ' num2str(params.octave_cost) ' ' ...
+                num2str(params.octave_jump_cost) ' ' num2str(params.voiced_unvoiced_cost) ' ' num2str(params.pitchlimits(2))]);
+        else
+            status = system(['"C:\Users\Public\Desktop\Praat.exe" --run get_pitch_tracks.praat "' pwd '" "temp_wav" ' ...
+                num2str(params.pitchlimits(1)) ' ' num2str(params.ptrack_max_candidates) ' ' params.ptrack_pitch_very_accurate_checkbox ' ' ...
+                num2str(params.ptrack_silence_thresh) ' ' num2str(params.ptrack_voicing_thresh) ' ' num2str(params.ptrack_octave_cost) ' ' ...
+                num2str(params.ptrack_octave_jump_cost) ' ' num2str(params.ptrack_voiced_unvoiced_cost) ' ' num2str(params.pitchlimits(2))]);
+        end
```

Comment:
```suggestion
        if ismac
            praatExePath = "/Applications/Praat.app/Contents/MacOS/Praat"
        else
            praatExePath = "C:\Users\Public\Desktop\Praat.exe";
        end
        praatCmd = sprintf('%s --run get_pitch_tracks.praat %s %s %d .... etc. etc. etc.
        status = system(praatCmd);
```
Even if you don't want to use sprintf, it's definitely better to split out the path to praat.exe as a variable, because that appears to be the only thing that differs between the two commands; this way we can just have one command. (But sprintf will save you from all those num2str's and also double-quotes within single-quotes!)

Reply:
Yes, I cleaned this up later this afternoon, at least to get rid of the redundancy though not using  sprint.

Reply:
yes i cleaned this up this afternoon, though not using sprintf